homeassistant/components/intellifire/config_flow.py

```diff
@@ -127,3 +141,42 @@ async def async_step_user(
             return await self.async_step_pick_device()
         LOGGER.debug("Running Step: manual_device_entry")
         return await self.async_step_manual_device_entry()
+
+    async def async_step_dhcp(self, discovery_info: DhcpServiceInfo) -> FlowResult:
+        """Handle DHCP Discovery."""
+
+        # Run validation logic on ip
+        host = discovery_info.ip
+
+        self._async_abort_entries_match({CONF_HOST: host})
+        try:
+            serial = await validate_host_input(host)
+        except (ConnectionError, ClientConnectionError):
+            return self.async_abort(reason="not_intellifire_device")
+
+        self._discovered_hosts.append(DiscoveredHostInfo(ip=host, serial=serial))
+
+        placeholders = {CONF_HOST: host, "serial": serial}
+        self.context["title_placeholders"] = placeholders
+
+        return await self.async_step_dhcp_confirm()
+
+    async def async_step_dhcp_confirm(self, user_input=None):
+        """Attempt to confirm."""
+
+        # Add the hosts one by one
+        host = self._discovered_hosts[0].ip
+        serial = self._discovered_hosts[0].serial
+
+        if user_input is None:
+            # Show the confirmation dialog
+            return self.async_show_form(
+                step_id="dhcp_confirm", description_placeholders={CONF_HOST: host}
+            )
+
+        await self.async_set_unique_id(serial)
+        self._abort_if_unique_id_configured(updates={CONF_HOST: host})
```

Comment:
Set this before moving to the `dhcp_confirm` step so they aren't offered it if its already set up.  Please add a test for this case as well